docs/jp/faqs.md

```diff
@@ -0,0 +1,258 @@
+# よくある質問
+
+## State が更新されない
+
+❔ **質問**: State を yield してもUI側が更新されません。何が原因ですか？
+
+💡 **答え**: もし Equatable を使っているならきちんと state のプロパティ一覧を props に渡してあげてください。
+
+✅ **正解**
+
+```dart
+abstract class MyState extends Equatable {
+    const MyState();
+}
+
+class StateA extends MyState {
+    final String property;
+
+    const StateA(this.property);
+
+    @override
+    List<Object> get props => [property]; // 全てのプロパティを props に渡す
+}
+```
+
+❌ **間違い**
+
+```dart
+abstract class MyState extends Equatable {
+    const MyState();
+}
+
+class StateA extends MyState {
+    final String property;
+
+    const StateA(this.property);
+
+    @override
+    List<Object> get props => [];
+}
+```
+
+```dart
+abstract class MyState extends Equatable {
+    const MyState();
+}
+
+class StateA extends MyState {
+    final String property;
+
+    const StateA(this.property);
+
+    @override
+    List<Object> get props => null;
+}
+```
+
+それともう一つ、毎回新しい　state のインスタンスを yield するようにしてください。
+
+✅ **正解**
+
+```dart
+@override
+Stream<MyState> mapEventToState(MyEvent event) async* {
+    // 常に新しい state インスタンスを yield する
+    yield state.copyWith(property: event.property);
+}
+```
+
+```dart
+@override
+Stream<MyState> mapEventToState(MyEvent event) async* {
+    final data = _getData(event.info);
+    // 常に新しい state インスタンスを yield する
+    yield MyState(data: data);
+}
+```
+
+❌ **間違い**
+
+```dart
+@override
+Stream<MyState> mapEventToState(MyEvent event) async* {
+    // state のプロパティのみ変更するのはだめ
+    state.property = event.property;
+    // 同じインスタンスの state を yield するのはだめ
+    yield state;
+}
+```
+
+## Equatable を使うのはどんな時？
+
+❔**質問**: Equatable はどんな時に使うべき？
+
+💡**答え**:
+
+```dart
+@override
+Stream<MyState> mapEventToState(MyEvent event) async* {
+    yield StateA('hi');
+    yield StateA('hi');
+}
+```
+
+`StateA`が`Equatable`を継承している上のような場合では一度しか state は変化しません（２回目の yield は無視される）。
+一般的には再描画を最低限にしコードを最適化したい場合は`Equatable`を使うべきです。
+もし同じ state を返して transition を発生させたい場合は`Equatable`は使うべきではありません。
+
+加えて`Equatable`を使うとテストに置いて特定のプロパティを持った state を predict できるのでテストが楽になります。
+
+```dart
+blocTest(
+    '...',
+    build: () => MyBloc(),
+    act: (bloc) => bloc.add(MyEvent()),
+    expect: [
+        MyStateA(),
+        MyStateB(),
+    ],
+)
+```
+
+`Equatable`なしでは上記のテストコードは通らず、下記のように書かなくてはなりません：
+
+```dart
+blocTest(
+    '...',
+    build: () => MyBloc(),
+    act: (bloc) => bloc.add(MyEvent()),
+    expect: [
+        isA<MyStateA>(),
+        isA<MyStateB>(),
+    ],
+)
+```
+
+## Bloc vs. Redux
+
+❔ **質問**: Bloc と Redux の違いは何？
+
+💡 **答え**:
+
+BLoC は下記のルールを元に設計されたデザインパターンです:
+
+1. Bloc の入力と出力はシンプルな Stream と Sink であるべき。
+2. 依存性は注入可能で、プラットフォームに依存しない。
+3. プラットフォームに分岐してはならない。
+4. 上記のルールを守っている限り実装方法はどのような形でも良い。
+
+UIのガイドラインは:
+
+1. 一つ一つの「それなりに複雑な」コンポーネントは Bloc を持つべき。
+2. コンポーネントは入力をありのまま送るべき。
+3. コンポーネントは出力を出来るだけありのまま表示するべき。
+4. 全ての分岐はシンプルに BloC の主力に関連するべき。
```

Comment:
I am not very confident about the translation here, because I feel like I don't fully understand the line here that says
"4. All branching should be based on simple BLoC boolean outputs.".

Could you elaborate on this and explain what exactly you mean by this sentence?

Reply:
This means you should avoid having complex logic to determine what to render. Instead, the UI's logic should be really simple like

```dart
if (state is X) return WidgetA();
if (state is Y) return WidgetB();
```

Let me know if that helps 👍

Reply:
Thank you! That helps a lot. Let me change the Japanese sentence there in a minute.